microsoft-edge/webview2/concepts/process-related-events.md

```diff
@@ -0,0 +1,589 @@
+---
+title: Handling process-related events in WebView2
+description: Process-related events in WebView2 and how apps can handle them for recovery and improved reliability.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 06/29/2023
+---
+# Handling process-related events in WebView2
+
+**Contents:**
+
+* [Events for processes that exited or failed](#events-for-processes-that-exited-or-failed)
+* [Handle process failures](#handle-process-failures)
+   * [Gather process failure details](#gather-process-failure-details)
+      * [Failure kind](#failure-kind)
+      * [Failure reason](#failure-reason)
+      * [Exit code, process description, and frames information](#exit-code-process-description-and-frames-information)
+      * [Gathering details](#gathering-details)
+   * [Handle unexpected exits for various types of processes](#handle-unexpected-exits-for-various-types-of-processes)
+      * [Types of processes](#types-of-processes)
+      * [The main browser process has exited unexpectedly](#the-main-browser-process-has-exited-unexpectedly)
+      * [A process rendering content in the WebView2 control has exited unexpectedly](#a-process-rendering-content-in-the-webview2-control-has-exited-unexpectedly)
+      * [The GPU process has exited unexpectedly](#the-gpu-process-has-exited-unexpectedly)
+      * [A utility process has exited unexpectedly](#a-utility-process-has-exited-unexpectedly)
+      * [Any other process has exited unexpectedly](#any-other-process-has-exited-unexpectedly)
+   * [Handle unresponsive renderers](#handle-unresponsive-renderers)
+* [Handle main browser process crashes](#handle-main-browser-process-crashes)
+   * [ProcessFailed, for main browser process crashes](#processfailed-for-main-browser-process-crashes)
+   * [BrowserProcessExited, for main browser process crashes](#browserprocessexited-for-main-browser-process-crashes)
+* [Handle main browser process exited](#handle-main-browser-process-exited)
+   * [BrowserProcessExited event](#browserprocessexited-event)
+   * [Clearing the user data folder](#clearing-the-user-data-folder)
+   * [Updating the WebView2 Runtime](#updating-the-webview2-runtime)
+   * [Restarting with a different environment configuration](#restarting-with-a-different-environment-configuration)
+   * [Clearing the auth cache](#clearing-the-auth-cache)
+* [See also](#see-also)
+
+
+WebView2 uses multiple processes to support the WebView2 controls in your application. Because these processes can exit during use, WebView2 provides the `CoreWebView2.ProcessFailed` and `CoreWebView2Environment.BrowserProcessExited` events for your application to react to different scenarios.  Use this document to learn how to use these events to react when these scenarios occur.
```

Comment:
for search-engine optimization, move this Paragraph 1 to above the top-of-article TOC.